diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/files/IndentationRule.kt

```diff
@@ -290,40 +291,60 @@ class IndentationRule(configRules: List<RulesConfig>) : DiktatRule(
         actualIndent: Int
     ) {
         val templateEntries = stringTemplate.getAllChildrenWithType(LITERAL_STRING_TEMPLATE_ENTRY)
-        templateEntries.asSequence().filterIndexed { index, templateEntry ->
+        val templateEntriesSize = templateEntries.size
+        var templateEntryFollowingNewline = false
+
+        templateEntries.forEachIndexed { index, templateEntry ->
             val text = templateEntry.text
-            val containsNewline = text.contains(NEWLINE)
 
-            if (containsNewline) {
+            when {
+                text.contains(NEWLINE) -> {
+                    /*
+                     * Set the flag.
+                     */
+                    templateEntryFollowingNewline = true
+
+                    /*
+                     * In real-life cases observed, whenever a `LITERAL_STRING_TEMPLATE_ENTRY`
+                     * _contains_ a newline character, it is _exactly_ a newline character.
+                     */
+                    check(text.length == 1) {
+                        val escapedText = text.replace(NEWLINE.toString(), "\\n")
+
+                        "A LITERAL_STRING_TEMPLATE_ENTRY at index $index contains extra characters in addition to the newline, " +
+                            "entry: \"$escapedText\", " +
+                            "string template: ${stringTemplate.text}"
+                    }
+                }
+
                 /*
-                 * In real-life cases observed, whenever a `LITERAL_STRING_TEMPLATE_ENTRY`
-                 * _contains_ a newline character, it is _exactly_ a newline character.
+                 * This is the last string template fragment which is usually followed
+                 * with the closing `"""` and the `.trimIndent()` or `.trimMargin(...)` call.
                  */
-                check(text.length == 1) {
-                    val escapedText = text.replace(NEWLINE.toString(), "\\n")
+                index == templateEntriesSize - 1 -> {
```

Comment:
to be fair - I would move this `- 1` logic to the `templateEntriesSize` initialisation

Reply:
Thanks!

Addressed with 1591d9ee597132e2f4d1892f643dc43c6b5f73d6.